Kitodo/src/main/java/org/kitodo/production/interfaces/activemq/TaskActionProcessor.java

```diff
@@ -0,0 +1,212 @@
+/*
+ * (c) Kitodo. Key to digital objects e. V. <contact@kitodo.org>
+ *
+ * This file is part of the Kitodo project.
+ *
+ * It is licensed under GNU General Public License version 3 or later.
+ *
+ * For the full copyright and license information, please read the
+ * GPL3-License.txt file that was distributed with this source code.
+ */
+
+package org.kitodo.production.interfaces.activemq;
+
+import java.io.IOException;
+import java.util.Date;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import javax.jms.JMSException;
+
+import org.kitodo.config.ConfigCore;
+import org.kitodo.config.enums.ParameterCore;
+import org.kitodo.data.database.beans.Comment;
+import org.kitodo.data.database.beans.Task;
+import org.kitodo.data.database.beans.User;
+import org.kitodo.data.database.enums.CommentType;
+import org.kitodo.data.database.enums.TaskEditType;
+import org.kitodo.data.database.enums.TaskStatus;
+import org.kitodo.data.database.exceptions.DAOException;
+import org.kitodo.data.elasticsearch.exceptions.CustomResponseException;
+import org.kitodo.data.exceptions.DataException;
+import org.kitodo.exceptions.ProcessorException;
+import org.kitodo.production.services.ServiceManager;
+import org.kitodo.production.services.data.TaskService;
+import org.kitodo.production.services.workflow.WorkflowControllerService;
+
+/**
+ * This is a web service interface to modify task status.
+ */
+public class TaskActionProcessor extends ActiveMQProcessor {
+
+    public static final String KEY_CORRECTION_TASK_ID = "correctionTaskId";
+    public static final String KEY_MESSAGE = "message";
+    public static final String KEY_TASK_ACTION = "action";
+    public static final String KEY_TASK_ID = "id";
+    private final TaskService taskService = ServiceManager.getTaskService();
+    private WorkflowControllerService workflowControllerService;
+
+    /**
+     * The default constructor looks up the queue name to use in kitodo_config.properties. If that is not configured and
+     * “null” is passed to the super constructor, this will prevent ActiveMQDirector.registerListeners() from starting
+     * this service.
+     */
+    public TaskActionProcessor() {
+        super(ConfigCore.getOptionalString(ParameterCore.ACTIVE_MQ_TASK_ACTION_QUEUE).orElse(null));
+        workflowControllerService = new WorkflowControllerService();
+    }
+
+    /**
+     * This is the main routine processing incoming messages. It gets the task id and the task action for processing.
+     * Every action has its own behavior so please read the comment on the action for more information.
```

Comment:
I am not sure if I missed something but where are these comments explaining each action? Do yo mean the exception messages?

Reply:
No i mean the JavaDoc of the the task action enum.